hs_core/tasks.py

```diff
@@ -167,23 +167,21 @@ def create_bag_by_irods(resource_id, istorage=None):
     from hs_core.hydroshare.utils import get_resource_by_shortkey
 
     res = get_resource_by_shortkey(resource_id)
-    try:
-        # we will create xml metadata files every time prior to creating the bag
-        # the assumption here is that bag is being created due to the resource
-        # modified (AVU)
-        create_bag_files(res, fed_zone_home_path=res.resource_federation_path)
-    except Exception as ex:
-        logger.error('Failed to create bag files. Error:{}'.format(ex.message))
-        return False
+    if istorage is None:
+        istorage = res.get_irods_storage()
 
-    is_exist = False
-    is_bagit_readme_exist = False
+    metadata_dirty = istorage.getAVU(res.root_path, 'metadata_dirty')
+    # if metadata has been changed, then regenerate metadata xml files
+    if metadata_dirty == "true":
+        try:
+            create_bag_files(res, fed_zone_home_path=res.resource_federation_path)
+        except Exception as ex:
+            logger.error('Failed to create bag files. Error:{}'.format(ex.message))
+            return False
 
     bag_full_name = 'bags/{res_id}.zip'.format(res_id=resource_id)
     if res.resource_federation_path:
-        if not istorage:
-            istorage = IrodsStorage('federated')
-        irods_bagit_input_path = os.path.join(res.resource_federation_path, resource_id)
+        irods_bagit_input_path = res.root_path

```

Comment:
@pkdash Thanks for giving this a try. It looks like bagit rule execution requires a full path for bagit input. In this case, I'd suggest to keep the original ```irods_bagit_input_path = os.path.join(res.resource_federation_path, resource_id)``` rather than replacing it with ```irods_bagit_input_path = res.root_path``` in the ```if``` block as you did currently since that just introduce an additional layer of conditional check with no benefit. After you finish this simple change, I will give a +1.